geoengine/datasets.py

```diff
@@ -95,7 +100,7 @@ class ValueOgrSourceDurationSpec(OgrSourceDuration):
 
     step: TimeStep
 
-    def to_dict(self) -> Dict[str, str]:
+    def to_dict(self) -> Dict[str, Any]:
```

Comment:
Is it really `Any`? You did some more concrete typedef below with `Union[str, Dict[str, str]]`. Maybe we can define a typedef that works.

Reply:
There are different possible approaches here. `Any` is used in this case, because the returned dictionary is composed of different types. It would also be possible to use something like this:

```suggestion
    def to_dict(self) -> Dict[str, Union[str, int, TimeStepGranularity]]:
```

This leaves the superclass signature as `def to_dict(self) -> Dict[str, Any]:`. If that should be narrowed down as well, I could add an implementation using `typing.Generics` and `typing.TapeVar` which adds a little bit more complexity to the class. This would be along the likes of this [stackoverflow answer](https://stackoverflow.com/a/71951574).

Reply:
An attempt to remove the need for `Any` in the base class: [036ff28](https://github.com/geo-engine/geoengine-python/pull/70/commits/036ff28b64a1f364e7d1b8abd16345eae923c9e6)